src/pool/mempool.rs

```diff
@@ -131,11 +140,12 @@ impl<SP: starknet::providers::Provider + Send + Sync + Clone + 'static> AccountM
 
             // If the balance is lower than the threshold, continue
             if balance < U256::from(u128::pow(10, 18)) {
+                accounts.remove(account_address);
```

Comment:
Isn't there a possibility of DoS here if each of the accounts ends up emptied successively. At the end as the `accounts` is not updated in the loop, if all the accounts are removed from this variable it is possible that there will be none left no? And even if in the meantime we refill the gaps by rebalancing they will not be able to integrate the list along the way as long as we do not restart the server no?

Reply:
you won't be able to lock the account which is what we want, so `lock_account` will return an error (see break condition on `accounts.is_empty()`. 

Then the outer loop will quick in as well and we will never be able to fetch an account.

If we rebalance, the accounts variable will be reset to `self.accounts.iter()` so it should be ok

Reply:
Aaaah yes got you here, should be ok then